docs/models/overrides.md

```diff
@@ -0,0 +1,36 @@
+Item Property Overrides
+=======================
+
+When you have an item for which you want to use different models depending on its state, you have a few options: use a blockstate JSON, several item models, or use item properties. When you use a blockstate JSON or several item models for an item, you can directly bind certain metadatas to certain variants/models, and with an `ItemMeshDefinition` you can use arbitrary code to map `ItemStack`s to variants/models. However, what if you need more flexibility? For example, a bow needs to know how long it has been pulled, which is most easily accessed as a property of the holding entity. Minecraft 1.8 introduced item property overrides. An item property assigns a certain `float` value to every ItemStack it is registered for, and vanilla item model definitions can use these values to define "overrides", where an item defaults to a certain model, but if an override matches, it overrides the model and uses another. The format of item models, including overrides, can be found on the [wiki]. In the example with bows, an item property is especially nice as properties can be continuous values. Instead of hardcoding only 4 slots for bow models, item properties allow you to define as many models as you want on the spectrum of the "pulling" property.
```

Comment:
Including custom baked models here doesn't really make sense. Those are interchangeable with all other models, and therefore they fall under the category of "several item models".